### PR TITLE
fix: form types and restaurant completions fixes

### PR DESCRIPTION
**Pull request by callumcurtis**

### CODE REVIEW

File: apps/nextjs/src/components/friends.tsx [hunk 1]
```diff
@@ -342,10 +342,11 @@ function AddFriendForm(props: {
     },
   });
 
-  const onSubmit = (data: z.infer<typeof AddFriendFormSchema>) => {
+  const onSubmit = (formInput: z.input<typeof AddFriendFormSchema>) => {
+    const formOutput = AddFriendFormSchema.parse(formInput);
     sendFriendRequest.mutate(
```
Comment (anormananderson) [inline apps/nextjs/src/components/friends.tsx:347]: I assume this is obvious but why do we need to parse formInput again if it is already type AddFriendFormSchema?

File: apps/nextjs/src/components/reviews.tsx [hunk 2]
```diff
@@ -85,25 +86,22 @@ export function CreateReviewForm() {
     },
   });
 
-  const onSubmit = (data: z.input<typeof ReviewFormSchema>) => {
+  const onSubmit = (formInput: z.input<typeof ReviewFormSchema>) => {
+    const formOutput = ReviewFormSchema.parse(formInput);
     createReview.mutate(
```
Comment (anormananderson) [inline apps/nextjs/src/components/reviews.tsx:91]: Same question here as in the other file

**Review by anormananderson — APPROVED** [review]

File: apps/nextjs/src/components/friends.tsx (hunk 1 above)
Comment (callumcurtis) [reply apps/nextjs/src/components/friends.tsx:347]: Good question - `formInput` is of type `z.input<...>` which refers to the input type of the `AddFriendFormSchema`, whereas we want its output. [You can find more info on Zod inputs and outputs here](https://zod.dev/#:~:text=In%20reality%20each%20Zod%20schema%20internally%20tracks%20two%20types%3A%20an%20input%20and%20an%20output). This is an annoying detail of using Zod + React Hook Form - on submit, the form forwards the `input` types of the Zod schema.

We could wrap React Hook Form to change this behavior.

File: apps/nextjs/src/components/reviews.tsx (hunk 2 above)
Comment (callumcurtis) [reply apps/nextjs/src/components/reviews.tsx:91]: Answered in original place

**Pull request merged**